src/WebJobs.Script.Analyzers/WebJobsAttributeAnalyzer.cs

```diff
@@ -0,0 +1,331 @@
+        // propInfo is a property on the attribute.
+        private void ValidateAttributeProperty(SyntaxNodeAnalysisContext context, Attribute attribute, PropertyInfo propInfo, AttributeArgumentSyntax attributeSyntax)
+        {
+            var value = propInfo.GetValue(attribute);
+            var propertyAttributes = propInfo.GetCustomAttributes();
+
+            // First validate [AutoResolve] and [AppSetting].
+            // Then do validators.
+            bool isAutoResolve = false;
+            bool isAppSetting = false;
+            MethodInfo validator = null;
+            Attribute validatorAttribute = null;
+
+            foreach (Attribute propertyAttribute in propertyAttributes)
+            {
+                // AutoResolve and AppSetting are exclusive.
+                // TODO: Does GetType() really have to be called three times?
+                if (propertyAttribute.GetType() == typeof(Microsoft.Azure.WebJobs.Description.AutoResolveAttribute))
+                {
+                    isAutoResolve = true;
+                }
+                if (propertyAttribute.GetType() == typeof(Microsoft.Azure.WebJobs.Description.AppSettingAttribute))
+                {
+                    isAppSetting = true;
+                }
+
+                if (validator == null)
+                {
+                    validator = propertyAttribute.GetType().GetMethod("Validate", new Type[] { typeof(object), typeof(string) });
+                    validatorAttribute = propertyAttribute;
+                }
+            }
+
+            // Now apply error checks in order.
+            if (isAutoResolve)
+            {
+                // Value should parse with { } and %%
+                try
+                {
+                    if (value is string valueStr)
+                    {
+                        var template = Microsoft.Azure.WebJobs.Host.Bindings.Path.BindingTemplate.FromString(valueStr);
+                        if (template.HasParameters)
+                        {
+                            // The validator runs after the { } and %% are substituted.
+                            // But {} and %% may be illegal characters, so we can't validate with them.
+                            // So skip validation.
+                            // TODO - could we have some "dummy" substitution so that we can still do validation?
+                            return;
+                        }
+                    }
+                }
+                catch (FormatException e)
+                {
+                    // Parse error
+                    var error = Diagnostic.Create(DiagnosticDescriptors.BadBindingExpressionSyntax, attributeSyntax.GetLocation(), propInfo.Name, value, e.Message);
+                    context.ReportDiagnostic(error);
+                    return;
+                }
+            }
+            else if (isAppSetting)
+            {
+                // TODO - validate the appsetting. In local.json? etc?
```

Comment:
Is there still code needed here?

Reply:
This was an empty section + comment copied over from Mike's code. I assume he meant to perform more validation here but wasn't able to get around to it. I left it as a reminder if we wanted to continue the work as an enhancement.